api/v1/testkube.yaml

```diff
@@ -612,8 +676,18 @@ components:
         executionResult:
           required: true
           description: result get from executor
-          $ref: '#/components/schemas/ExecutionResult'
+          $ref: "#/components/schemas/ExecutionResult"
 
+    Artifact:
+      type: object
+      description: API server artifact
+      properties:
+        name:
+          type: string # https://api-server/exec_id/subdirectory/file_name
+          description: artifact file path
+        size:
```

Comment:
can we provide some metadata on content-type? (we can always guess from the file extension) - this would help clients with further processing... i.e. if they know the file is a pdf, avi, png, etc...

Reply:
minio will return full name with extension

Reply:
Just like this
![image](https://user-images.githubusercontent.com/8327751/140296631-a9431524-32f6-4ed7-94d6-7fcf847d55c4.png)

Reply:
so you're saying we should let the client decide based on extension only? I'm thinking that other executors might provide this metadata without the client having to guess... but we can perhaps cross that bridge when we get there....

Reply:
I tried to pull content-type (MIME) from minio but when listing objects from a bucket you don't get information.
Below is everything that minio returns about a single object:
```

(minio.ObjectInfo) {
 ETag: (string) (len=32) "7f932a200f44503cebd0c2ceeed8efb5",
 Key: (string) (len=7) "main.go",
 LastModified: (time.Time) 2021-11-04 10:32:32.845 +0000 UTC,
 Size: (int64) 471,
 ContentType: (string) "",
 Expires: (time.Time) 0001-01-01 00:00:00 +0000 UTC,
 Metadata: (http.Header) <nil>,
 UserMetadata: (minio.StringMap) <nil>,
 UserTags: (map[string]string) <nil>,
 UserTagCount: (int) 0,
 Owner: (minio.Owner) {
  XMLName: (xml.Name) {
   Space: (string) (len=39) "http://s3.amazonaws.com/doc/2006-03-01/",
   Local: (string) (len=5) "Owner"
  },
  DisplayName: (string) (len=64) "02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4",
  ID: (string) (len=5) "minio"
 },
 Grant: ([]minio.Grant) <nil>,
 StorageClass: (string) (len=8) "STANDARD",
 IsLatest: (bool) false,
 IsDeleteMarker: (bool) false,
 VersionID: (string) "",
 ReplicationStatus: (string) "",
 Expiration: (time.Time) 0001-01-01 00:00:00 +0000 UTC,
 ExpirationRuleID: (string) "",
 Restore: (*minio.RestoreInfo)(<nil>),
 Err: (error) <nil>
}
```